testing/integration/src/tester.rs

```diff
@@ -0,0 +1,227 @@
+        let (sys_code_cid, init_code_cid, accounts_code_cid) =
+            fetch_builtin_code_cid(&blockstore, &manifest_data_cid, manifest_version)?;
+
+        // Initialize state tree
+        let mut state_tree = StateTree::new(blockstore, stv).map_err(anyhow::Error::from)?;
+
+        // Insert an empty HAMT.
+        let empty_cid = Hamt::<_, String>::new_with_bit_width(state_tree.store(), 5)
+            .flush()
+            .unwrap();
+
+        // Deploy init and sys actors
+        let sys_state = system_actor::State { builtin_actors };
+        set_sys_actor(&mut state_tree, sys_state, sys_code_cid)?;
+
+        let init_state = init_actor::State {
+            address_map: empty_cid,
+            next_id: 100,
+            network_name: "test".to_owned(),
+        };
+        set_init_actor(&mut state_tree, init_code_cid, init_state)?;
+
+        Ok(Tester {
+            nv,
+            builtin_actors,
+            blockstore: None,
+            executor: None,
+            code_cids: vec![],
+            state_tree,
+            accounts_code_cid,
+        })
+    }
+
+    /// Creates new accounts in the testing context
+    pub fn create_account<const N: usize>(&mut self) -> Result<[Account; N]> {
+        // Create accounts.
+        put_secp256k1_accounts(&mut self.state_tree, self.accounts_code_cid)
+    }
+
+    /// Set a new state in the state tree
+    pub fn set_state<S: ser::Serialize>(&mut self, state: &S) -> Result<Cid> {
+        // Put state in tree
+        let state_cid = self.state_tree.store().put_cbor(state, Code::Blake2b256)?;
+
+        Ok(state_cid)
+    }
+
+    /// Set a new at a given address, provided with a given token balance
+    pub fn set_actor_from_bin(
+        &mut self,
+        wasm_bin: &[u8],
+        state_cid: Cid,
+        actor_address: Address,
+        balance: TokenAmount,
+    ) -> Result<()> {
+        // Register actor address
+        self.state_tree
+            .register_new_address(&actor_address)
+            .unwrap();
+
+        // Put the WASM code into the blockstore.
+        let code_cid = put_wasm_code(self.state_tree.store(), wasm_bin)?;
+
+        // Add code cid to list of deployed contract
+        self.code_cids.push(code_cid);
+
+        // Initialize actor state
+        let actor_state = ActorState::new(code_cid, state_cid, balance, 1);
+
+        // Create actor
+        self.state_tree
+            .set_actor(&actor_address, actor_state)
+            .map_err(anyhow::Error::from)?;
+
+        Ok(())
+    }
+
+    /// Sets the Machine and the Executor in our Tester structure.
+    pub fn instantiate_machine(&mut self) -> Result<()> {
+        // First flush tree and consume it
+        let state_root = self
+            .state_tree
+            .flush()
+            .map_err(anyhow::Error::from)
+            .context(FailedToFlushTree)?;
+
+        let blockstore = self.state_tree.store();
+
+        self.blockstore = Some(blockstore.clone());
```

Comment:
You are completly right, I missed the opportunity to access `Machine::blockstore()`. I updated the behaviour so that we access blockstore as state before instantiating and the `Machine`'s once instantiated.